packages/e2e/test/artillery/wallet-restoration/queries.ts

```diff
@@ -0,0 +1,14 @@
+/**
+ * Query randomized distinct addresses from db associated with users who are staking
+ */
+export const findAddressesWithRewardsHistory = `
+  SELECT * FROM (
+    SELECT DISTINCT txOut.address as address, sa.view as stake_address
+    FROM public.delegation d
+      LEFT JOIN public.tx_out txOut on
+    d.addr_id = txOut.stake_address_id
+      LEFT JOIN public.stake_address sa on
+    txOut.stake_address_id = sa.id
+  ) distinct_addresses
+  ORDER BY RANDOM() LIMIT $1
```

Comment:
The spec asks for:

> Number of transactions > 10 and less than 200

I don't think we actually need to restrict the query to addresses with greater than 10 transactions, but maybe we should just be ordering by tx count, which will make the test more consistent and ensure the significant wallets are selected.

Reply:
I agree wee don't need to restrict the transactions count per address, might be just the ordering as you suggested @rhyslbw

Reply:
I think to simulate something close to real usage we need a good distribution of # of tx'es. There are a few issues with sorting by tx count:
- doesn't easily scale across multiple workers - you need to share some state, or at least know  the worker number/index to do pagination in order to not select the same wallets on every worker.
- it would probably result in addresses that all have the same # of transactions (depends on number of addresses selected of course).

Reply:
In a real-world scenario is more likely N number of users to want to restore their wallets at the same time with different number of transactions each

Reply:
Fair points. If we need a consistent test scenario later on we can always extend this to allow for a consistent pre-test results set.